package.json

```diff
@@ -151,6 +152,7 @@
     "babel-runtime": "^6.20.0",
     "copy-webpack-plugin": "^4.0.1",
     "css-loader": "^0.26.1",
+    "doctoc": "^1.3.0",
```

Comment:
doctoc adds a lot of dependencies, do we really want this?
http://npm.anvaka.com/#/view/2d/doctoc

Reply:
If it's used to generate the ToC it should be included in the development dependencies like `webpack` and `standard` and everything else we use to generate code from or for the repository.

https://github.com/hackmdio/hackmd/blob/5ce8f40eac7fccfcfca175cf79ccfa86a84bc113/package.json#L144-L172

Are you worried about the production machines who run HackMD and need to install` doctoc` because `webpack` and other dependencies are needed for the build process?

Reply:
webpack is used by everyone who runs their own server
standard is run by evyone changing any line of javascript
doctoc would only be needed for those updating the readme. I wonder if this would not be a candidate for `optionalDependencies`?

Reply:
I wouldn't say that it's an optionalDependency since they are installed by `npm install` if you don't specify `--no-optional`.

The devDependency description says:

> If someone is planning on downloading and using your module in their program, then they probably don't want or need to download and build the external test or documentation framework that you use.

`doctoc` sounds like a documentation tool to me.

No matter if it is or not, I think when we want to keep the comments about doctoc, we should add it to the dependency list. If not, we should remove the comments.

@PeterDaveHello is your `doctoc` installed globally or did you install it in the project scope?

Reply:
I use it in the project scope, not everyone knows where it comes from, so the dependency can help anyone to fetch it without the issue.